front/lib/utils.ts

```diff
@@ -96,3 +96,20 @@ export const communityApps = [
     visibility: "public",
   },
 ];
+
+export const isValidUrl = (urlString: string) => {
```

Comment:
didn't want to add a lib just for that, so i stole a regex

Reply:
I think node has some utility? Maybe https://nodejs.org/api/url.html

Reply:
(You probably already have this but we also want to standardize the URL with protocol if not present)

Reply:
I use that, but for standardising only at the end. I don't think we can really rely on that for validation:
<img width="482" alt="Screenshot 2023-04-21 at 11 24 03" src="https://user-images.githubusercontent.com/14199823/233599532-57187c43-b932-410c-bdbd-924b638d0268.png">

Basically, the only thing this tool seems to care about is if there is `http[s]://` at the beginning of the string. I d

Reply:
Welll.... this is a valid URL :)

Reply:
But it's fair to not accept internal URLs here without any path behind.

Can you test something invalid at the path level?

Maybe we use URL and then we check that host has a TLD which will be less reliant on the regexp (which I'm pretty sure we can break with strange query parameters vs the battle-tested URL?)

Reply:
Ok cool -- Happy to do new URL() + check if host ends with a TLD (any 1+ length string).

> Can you test something invalid at the path level? 

Can't get it to fail. What would be an invalid path ?

Reply:
moved to node's URL only (on the server only since there is no longer a UI part): we require the protocol to be present, and it has to be either `http` or `https`. 
I don't check for a TLD, because that would require a list of valid TLDs I think (`www.hello.com`, `hello.com`, `www.hello` -> hard to tell which one isn't OK)